MiniSimTests/DeviceParserTests.swift

```diff
@@ -105,18 +108,73 @@ class DeviceParserTests: XCTestCase {
     XCTAssertEqual(devices[0].identifier, "mock_adb_id_for_Pixel_3a_API_30_x86")
     XCTAssertTrue(devices[0].booted)
     XCTAssertEqual(devices[0].platform, .android)
+    XCTAssertEqual(devices[0].type, .virtual)
 
     XCTAssertEqual(devices[1].name, "Pixel_4_API_29")
     XCTAssertEqual(devices[1].identifier, "mock_adb_id_for_Pixel_4_API_29")
     XCTAssertTrue(devices[1].booted)
     XCTAssertEqual(devices[1].platform, .android)
+    XCTAssertEqual(devices[1].type, .virtual)
 
     XCTAssertEqual(devices[2].name, "Nexus_5X_API_28")
     XCTAssertEqual(devices[2].identifier, nil)
     XCTAssertFalse(devices[2].booted)
     XCTAssertEqual(devices[2].platform, .android)
+    XCTAssertEqual(devices[2].type, .virtual)
   }
 
+  func testAndroidPhysicalDeviceParser() {
+    let parser = AndroidPhysicalDeviceParser()
+    let emptyInput = """
+        List of devices attached
+
+        """
+
+    var devices = parser.parse(emptyInput)
+    XCTAssertEqual(devices.count, 0)
+
+    let singleEmulatorInput = """
+      List of devices attached
+      emulator-5554          device product:sdk_gphone64_arm64 model:sdk_gphone64_arm64 device:emu64a transport_id:3
+
+      """
+
+    devices = parser.parse(singleEmulatorInput)
+    XCTAssertEqual(devices.count, 0)
+
+    let singlePhysicalDeviceInput = """
+      List of devices attached
+      RFCWA0FXXXX            device 0-1 product:a34xdxx model:SM_A346E device:a34x transport_id:5
+
+      """
+
+    devices = parser.parse(singlePhysicalDeviceInput)
+    XCTAssertEqual(devices.count, 1)
+
+    XCTAssertEqual(devices[0].name, "SM_A346E")
+    XCTAssertEqual(devices[0].identifier, "RFCWA0FXXXX")
+    XCTAssertTrue(devices[0].booted)
+    XCTAssertEqual(devices[0].platform, .android)
+    XCTAssertEqual(devices[0].type, .physical)
+
+    let mixedInput = """
+      List of devices attached
+      emulator-5554          device product:sdk_gphone64_arm64 model:sdk_gphone64_arm64 device:emu64a transport_id:3
+      RFCWA0FXXXX            device 0-1 product:a34xdxx model:SM_A346E device:a34x transport_id:5
+
+      """
+
+    devices = parser.parse(mixedInput)
+    XCTAssertEqual(devices.count, 1)
+
+    XCTAssertEqual(devices[0].name, "SM_A346E")
+    XCTAssertEqual(devices[0].identifier, "RFCWA0FXXXX")
+    XCTAssertTrue(devices[0].booted)
+    XCTAssertEqual(devices[0].platform, .android)
+    XCTAssertEqual(devices[0].type, .physical)
+  }
+
+
```

Comment:
Please run `swiftlint lint --fix ./` in the root to fix these linting issues

Reply:
done